doc/train_parameters.md

```diff
@@ -0,0 +1,155 @@
+| max_steps        | train.max_steps       | integer       | None          |                                                                                                                             |
+| eval steps       | eval.steps            | integer       | None          |                                                                                                                             |
+| start delay secs | eval.start_delay_secs | integer       | 120           |                                                                                                                             |
+| throttle_secs    | eval.throttle_secs    | integer       | 600           |                                                                                                                             |
+
+### COLUMN FOR
+The expressions in `COLUMN` contains the `Feature Columns` information. There could be multiple `COLUMN` blocks in the SQL. 
+
+The value of `FOR` keyword represents which parameter of the constructor method the `feature columns` assigning to.
+
+For example, the following SQL
+```sql
+...
+TRAIN
+	DNNLinearCombinedClassifier
+WITH
+	...
+COLUMN
+	NUMERIC(...) FOR linear_feature_columns
+COLUMN
+	BUCKET(...) FOR dnn_feature_columns
+...
+```
+will be translated to 
+```python
+	estimator = tf.estimator. DNNLinearCombinedClassifier(
+		linear_feature_columns = [tf.feature_column.numeric(...)],
+		dnn_feature_columns = [tf.feature_column.bucket(...)],
+		...
+	)
+```
+
+They were not only translated into [TensorFlow Feature Columns](https://www.tensorflow.org/guide/feature_columns), but also contains the encoding information of data such as `DENSE` or `SPARSE`. 
+
+For example, the following SQL
+```sql
+SELECT 
+...
+COLUMN
+	DENSE(c1, ...)
+	SPARSE(c2, ...)
+```
+represents that the `c1` field is the dense format and the `c2` field is the sparse format.
+
+<b>Here is a list of supported expression:</b>
+
+| expression | arguments of expression                                          | example                            |
+|------------|------------------------------------------------------------------|------------------------------------|
+| dense      | 1. field name (str) <br> 2. dense shape (integer) <br> 3. separator (str)  | dense(c1, 100, comma)              |
+| sparse     | 1. field name (str) <br> 2. sparse shape (integer) <br> 3. separator (str) | sparse(c2, 10000, comma)           |
+| numeric    | 1. field name (str) <br> 2. shape (integer)                           | numeric(c1, 100)                   |
+| bucket     | 1. key (numeric) <br> 2. bucket size (integer)                        | bucket(numeric(c1, 100), 20)       |
+| cat_id     | 1. field name (str) <br> 2. bucket size (integer)                     | cat_id(c2, 10000)                  |
```

Comment:
cat_id -> categorize

and, why always need bucket size, the column may be enumerated, like `Male, Female, NULL`

Reply:
In the original design, the `cat_id` is short for `categorical_column_with_identity ` which needs `bucket size`. Does renaming to `categorize` means it has to support all the APIs of `categorical_column_*`?